### PR TITLE
Resolving clippy warning

### PR DESCRIPTION
**Pull request by allaboutevemirolive**
I've cut the Clippy warnings in half, dropping from 57 to 28. The remaining warnings likely require changes to variable names.

@BurntSushi

### CODE REVIEW

File: src/dfa.rs [hunk 1]
```diff
@@ -494,7 +494,7 @@ impl Builder {
             matches: vec![vec![]; num_match_states],
             matches_memory_usage: 0,
             pattern_lens: nnfa.pattern_lens_raw().to_vec(),
-            prefilter: nnfa.prefilter().map(|p| p.clone()),
+            prefilter: nnfa.prefilter().cloned(),
```
Comment (BurntSushi) [inline src/dfa.rs:497]: I agree with this one.

File: src/packed/api.rs [hunk 2]
```diff
@@ -676,7 +676,7 @@ impl<'s, 'h> Iterator for FindIter<'s, 'h> {
         if self.span.start > self.span.end {
             return None;
         }
-        match self.searcher.find_in(&self.haystack, self.span) {
+        match self.searcher.find_in(self.haystack, self.span) {
```
Comment (BurntSushi) [inline src/packed/api.rs:679]: I agree with this one.

File: src/nfa/contiguous.rs [hunk 3]
```diff
@@ -821,7 +821,7 @@ impl<'a> State<'a> {
 
     /// Return an iterator over every explicitly defined transition in this
     /// state.
-    fn transitions<'b>(&'b self) -> impl Iterator<Item = (u8, StateID)> + 'b {
+    fn transitions(&self) -> impl Iterator<Item = (u8, StateID)> + '_ {
```
Comment (BurntSushi) [inline src/nfa/contiguous.rs:824]: I agree with this one.

File: src/packed/pattern.rs [hunk 4]
```diff
@@ -211,7 +211,7 @@ pub(crate) struct Pattern<'a>(&'a [u8]);
 impl<'a> fmt::Debug for Pattern<'a> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.debug_struct("Pattern")
-            .field("lit", &String::from_utf8_lossy(&self.0))
+            .field("lit", &String::from_utf8_lossy(self.0))
```
Comment (BurntSushi) [inline src/packed/pattern.rs:214]: I agree with this one.

File: src/packed/pattern.rs [hunk 5]
```diff
@@ -224,7 +224,7 @@ impl<'p> Pattern<'p> {
 
     /// Returns the bytes of this pattern.
     pub(crate) fn bytes(&self) -> &[u8] {
-        &self.0
+        self.0
```
Comment (BurntSushi) [inline src/packed/pattern.rs:227]: I agree with this one.

File: src/util/prefilter.rs [hunk 6]
```diff
@@ -330,7 +330,7 @@ struct Packed(packed::Searcher);
 impl PrefilterI for Packed {
     fn find_in(&self, haystack: &[u8], span: Span) -> Candidate {
         self.0
-            .find_in(&haystack, span)
+            .find_in(haystack, span)
```
Comment (BurntSushi) [inline src/util/prefilter.rs:333]: I agree with this one.

File: src/util/prefilter.rs [hunk 7]
```diff
@@ -549,7 +543,7 @@ impl RareBytesBuilder {
             let (mut bytes, mut len) = ([0; 3], 0);
             for b in 0..=255 {
                 if builder.rare_set.contains(b) {
-                    bytes[len] = b as u8;
+                    bytes[len] = b;
```
Comment (BurntSushi) [inline src/util/prefilter.rs:546]: I agree with this one.

File: src/util/prefilter.rs [hunk 8]
```diff
@@ -604,7 +598,7 @@ impl RareBytesBuilder {
             self.available = false;
             return;
         }
-        let mut rarest = match bytes.get(0) {
+        let mut rarest = match bytes.first() {
```
Comment (BurntSushi) [inline src/util/prefilter.rs:601]: Agree.

File: src/util/prefilter.rs [hunk 9]
```diff
@@ -835,7 +829,7 @@ impl StartBytesBuilder {
         if self.count > 3 {
             return;
         }
-        if let Some(&byte) = bytes.get(0) {
+        if let Some(&byte) = bytes.first() {
```
Comment (BurntSushi) [inline src/util/prefilter.rs:832]: Agree.

**Review by BurntSushi — CHANGES_REQUESTED** [review]
I don't really agree with most of these changes. I don't feel terribly strong about most of them, but generally prefer to not accept most of these. This is why I don't have Clippy setup to run. (I realize I could pick and choose which lints I want to enable, but this is a lot of work to do across the number of repos I maintain and just ultimately isn't worth it for me.)

I marked the changes that I agree with.

**Comment by allaboutevemirolive** [comment]
Is this okay with you? @BurntSushi

**Review by BurntSushi — APPROVED** [review]
Thanks!

**Pull request merged**